lib/watir/elements/select.rb

```diff
@@ -55,8 +62,11 @@ def select_all(*str_or_rx)
     #
 
     def select!(*str_or_rx)
-      results = str_or_rx.flatten.map { |v| select_by!(v, :single) }
-      results.first
+      if str_or_rx.size > 1 || str_or_rx.first.is_a?(Array)
+        str_or_rx.flatten.map { |v| select_by! v, :multiple }.first
+      else
+        str_or_rx.flatten.map { |v| select_by! v, :first }.first
```

Comment:
Is this supposed to be changing the second argument from `:single` to `:first`?

If I am looking at the related JS snippets correctly, they are only stopping the selection if the second argument is "single". So with this change `#select!` will always be selecting all instead of just the first.

Reply:
Ah, I actually couldn't find where that was  being used to know what the right keyword was to put there. Good catch.